misc/package/Data Files/MWSE/core/lib/DependencyManager/v1/init.lua

```diff
@@ -0,0 +1,347 @@
+                    end
+                end
+            end
+            return false
+        end,
+    },
+
+    {
+        pattern = "=",
+        callback = function(major, minor, patch, targetMajor, targetMinor, targetPatch)
+            if major == targetMajor and minor == targetMinor and patch == targetPatch then
+                return true
+            end
+            return false
+        end
+    },
+}
+
+
+
+function DependencyManager:doVersionCheck(dependency, version)
+    if dependency.version == nil then
+        --no version required
+        return true
+    end
+    if version == nil then
+        -- version required but none found
+        return false, string.format("Could not find version field in lua file: %s", dependency.luaFile)
+    end
+    self.logger:debug("Version: %s", version)
+    local major, minor, patch = string.match(version, "(%d+)%.(%d+)%.(%d+)")
+    self.logger:debug("Split version: Major: %s, Minor: %s, Patch: %s", major, minor, patch)
+    if not ( major and minor and patch ) then
+        return false, string.format("Invalid version: %s", version)
+    end
+    local targetMajor, targetMinor, targetPatch = string.match(dependency.version, "(%d+)%.(%d+)%.(%d+)")
+
+    local thisOperator
+    --find one of possible operators at start of string
+    self.logger:debug("Required version: %s", dependency.version)
+    for _, operator in ipairs(self.operators) do
+        self.logger:debug("operator pattern: %s", operator.pattern)
+        if string.startswith(dependency.version, operator.pattern) then
+            thisOperator = operator
+            break
+        end
+    end
+    if not thisOperator then
+        self.logger:debug("No operator found, setting to '='")
+        thisOperator = self.operators[3]
+    end
+
+    self.logger:debug("Operator: %s, Target: Major: %s, Minor: %s, Patch: %s", thisOperator.pattern, targetMajor, targetMinor, targetPatch)
+    if not ( targetMajor and targetMinor and targetPatch ) then
+        return false, string.format("Invalid version: %s", dependency.version)
+    end
+    if not thisOperator.callback(major, minor, patch, targetMajor, targetMinor, targetPatch) then
+        return false, string.format("Version needed: %s; Current version: %s", dependency.version, version)
+    end
+    return true
+end
+
+function DependencyManager:doMetadataCheck(dependency, metadataPath)
+    local metadataFile = io.open(metadataPath, "r")
+    if not metadataFile then
+        self.logger:error("Could not find dependency metadata file: %s", metadataPath)
+        return false, string.format("Could not find dependency metadata file: %s", metadataPath)
+    else
+        local metadata = json.decode(metadataFile:read("*all"))
+        if not metadata then
+            self.logger:error("Could not parse dependency metadata file: %s", metadataPath)
+            return false, string.format("Could not parse dependency metadata file: %s", metadataPath)
+        end
```

Comment:
Since we now have a [toml ](https://mwse.github.io/MWSE/apis/toml/)parser can you swap it instead of json now?

Reply:
Done